cmd/agent/model.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Reborndb Org. All Rights Reserved.
+// Licensed under the MIT (MIT-LICENSE.txt) license.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/go-zookeeper/zk"
+	"github.com/ngaut/zkhelper"
+)
+
+type agentInfo struct {
+	ID   string
+	Addr string `json:"addr"`
+	PID  int    `json:"pid"`
+}
+
+func agentPath() string {
+	return fmt.Sprintf("/zk/reborn/db_%s/agent", globalEnv.ProductName())
+}
+
+func addAgent(a *agentInfo) error {
+	basePath := agentPath()
+
+	zkhelper.CreateRecursive(globalConn, basePath, "", 0, zkhelper.DefaultDirACLs())
+
+	contents, err := json.Marshal(a)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	_, err = globalConn.Create(path.Join(basePath, a.ID), contents, zk.FlagEphemeral, zkhelper.DefaultFileACLs())
+	return errors.Trace(err)
+}
+
+func getAgents() ([]*agentInfo, error) {
+	basePath := agentPath()
+
+	children, _, err := globalConn.Children(basePath)
+	if err != nil && !zkhelper.ZkErrorEqual(err, zk.ErrNoNode) {
```

Comment:
Only refer the old code here

```
func ProxyList(coordConn zkhelper.Conn, productName string, filter func(*ProxyInfo) bool) ([]ProxyInfo, error) {
	ret := make([]ProxyInfo, 0)
	root := GetProxyPath(productName)
	proxies, _, err := coordConn.Children(root)
	if err != nil && !zkhelper.ZkErrorEqual(err, zk.ErrNoNode) {
		return nil, errors.Trace(err)
	}
```

We will continue only the ErrNoNode error and nil error, and stop for other errors.